src/components/TodoItem.tsx

```diff
@@ -0,0 +1,140 @@
+import classNames from 'classnames';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  title: string;
+  completed: boolean;
+  id: number;
+  isLoading: boolean;
+  deleteTodos: (id: number) => void;
+  updateTodo: (id: number, data: Partial<Todo>) => void;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  title,
+  completed,
+  id,
+  isLoading = false,
+  deleteTodos,
+  updateTodo,
+}) => {
+  const [loader, setLoader] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const cancelEditing = useCallback(() => {
+    setNewTitle(title);
+    setIsEditing(false);
+  }, [title]);
+
+  useEffect(() => {
+    setLoader(isLoading);
+
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+    }
+
+    const handleKeyUp = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        cancelEditing();
+      }
+    };
+
+    window.addEventListener('keyup', handleKeyUp);
+
+    return () => {
+      window.removeEventListener('keyup', handleKeyUp);
+    };
```

Comment:
This is redundant logic. Please use appropriate events on input (onBlur and onKeyUp) to handle this logic.